valeriano-manassero/trino/templates/configmap-coordinator.yaml

```diff
@@ -76,13 +76,15 @@ data:
     resource-groups.config-file={{ .Values.config.general.path }}/resource-groups/{{ .Values.resourceGroups.configFile | default "resource-groups.json" }}
 {{- end }}{{- end }}
 
-{{- if .Values.groupProvider }}{{- if .Values.groupProvider.type }}
+{{- if .Values.groupProvider }}{{- if .Values.groupProvider.type }}{{- if .Values.groupProvider.name }}
```

Comment:
This feels like overkill checking. Can we just check if groupProvider is there and then error out if these properties aren't present? You can use the `required` template function for this - there's an [example in our Trino chart](https://github.com/CybercentreCanada/hogwarts-trino-deployment/blob/c6410bd07a22be54a7d15aad9d1f7b9502e0cdd0/templates/configmap-coordinator.yaml#L34).

Reply:
Good idea, I did it for groupProvider.name but I didn't touch .type. I can't require .type here cause it's not actually being used and other configs all over the chart are also checking for *.type so I'd rather not make that change all over the place.

Reply:
Ok, but if `type` isn't being used, that bets the question... Why is it part of your condition at all? Do we care if it's there?

Reply:
So I think the idea is that we care because we don't want the config to exist and refer to a file that doesn't exist (which it won't if .type is missing because the pvc or configmap will be missing).

Reply:
Ok, but is there a valid use case where `groupProvider` is a non-empty object and we want this section of the template to be omitted rather than generating an error?

Reply:
I don't think so!

Reply:
```code
{{- if .Values.groupProvider }}
{{- required "groupProvider requires a valid 'type' ('configmap' or 'pvc')." .Values.groupProvider.type }}
  group-provider.properties: |
    group-provider.name={{ required "groupProvider requires a valid 'name'." .Values.groupProvider.name }}
{{- if eq .Values.groupProvider.name "file" }}
    file.refresh-period={{ .Values.groupProvider.refreshPeriod | default "5s" }}
    file.group-file={{ .Values.config.general.path }}/group-provider/{{ .Values.groupProvider.configFile | default "groups.txt" }}
{{- end }}
{{ .Values.groupProvider.customProperties | indent 4 }}
{{- end }}
```
Something like so? I don't think I've ever done something like this in a chart but I gave it a quick test and it works no problem.

Reply:
As discussed offline, you are following an established pattern in this chart. So I'm approving this. 😄